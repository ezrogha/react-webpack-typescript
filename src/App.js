import React, { useState } from 'react';
import { hot } from 'react-hot-loader/root';
import RT from './assets/RT.png'

const App = () => {
    const [state, setState] = useState("Timbi")
    return (
        <div>
            <h1 style={{ color: "red" }}>HELLO FROM APpgerfwP{state}</h1>
            <img src={RT} />
            <button onClick={() => setState("Rogha")}>CHANGE</button>
        </div>
    )
}

export default hot(App);