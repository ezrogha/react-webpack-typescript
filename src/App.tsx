import React, { useState } from 'react';
import RT from './assets/RT.png'
import { hola } from './hj';

type HeaderProps = { name: string }

const Header = ({name}: HeaderProps) => (<h2>{name}</h2>)

const App = () => {
    const [state, setState] = useState("Timbi")
    return (
        <div>
            <h1>Hello, World!</h1>
            <h2>{hola()}</h2>
            <Header name="Title" />
            <h1 style={{ color: "red" }}>HOLA FROM APpgerfwP{state}</h1>
            <img src={RT} />
            <button onClick={() => setState("Rogha")}>CHANGE</button>
        </div>
    )
}

export default App;