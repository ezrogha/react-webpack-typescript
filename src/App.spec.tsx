import React from 'react'
import { render } from '@testing-library/react'
import App from "./App"

describe('App.js', () => {
    test('should be true', () => {
        expect(true).toEqual(true)
    })

    test('should work', () => {
        const { getByText, container } = render(<App />)
        expect(getByText("CHANGE")).toBeInTheDocument()
        expect(container).toMatchSnapshot()
    })

    test('should ', () => {
        const { queryByText } = render(<App />)
        expect(queryByText("CHANGE")).toBeTruthy()
    })
    
})
