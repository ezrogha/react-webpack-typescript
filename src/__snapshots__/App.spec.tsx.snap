// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.js should work 1`] = `
<div>
  <div>
    <h1>
      Hello, World!
    </h1>
    <h2>
      BRRRR
    </h2>
    <h2>
      Title
    </h2>
    <h1
      style="color: red;"
    >
      HOLA FROM APpgerfwP
      Timbi
    </h1>
    <img
      src=""
    />
    <button>
      CHANGE
    </button>
  </div>
</div>
`;
